src/bin/wasm-tools/objdump.rs

```diff
@@ -36,15 +38,27 @@ impl Opts {
                 CodeSectionStart { range, count, .. } => printer.section_raw(range, count, "code"),
                 CodeSectionEntry(_) => {}
 
-                ComponentTypeSection(_) => todo!("component-model"),
-                ComponentImportSection(_) => todo!("component-model"),
-                ComponentFunctionSection(_) => todo!("component-model"),
-                ModuleSection { .. } => todo!("component-model"),
-                ComponentSection { .. } => todo!("component-model"),
-                InstanceSection(_) => todo!("component-model"),
-                ComponentExportSection(_) => todo!("component-model"),
-                ComponentStartSection { .. } => todo!("component-model"),
-                AliasSection(_) => todo!("component-model"),
+                ComponentTypeSection(s) => printer.section(s, "types"),
+                ComponentImportSection(s) => printer.section(s, "imports"),
+                ComponentFunctionSection(s) => printer.section(s, "functions"),
+                ModuleSection { range, .. } => { 
+                    printer.section_raw(range, 1, "module");
+                    if let Some(space) = printer.indices.last_mut() {
+                        space.inc_module();
+                    }
+                },
+
+                ComponentSection { range, .. } => {
+                    let mut index_space = IndexSpace::default();
+                    index_space.inc_component();
+                    printer.indices.push(index_space);
+                    printer.section_raw(range, 1, "component")
+                },
+
+                InstanceSection(s) => printer.section(s, "instances"),
+                ComponentExportSection(s) => printer.section(s, "exports"),
+                ComponentStartSection(_) => {},
```

Comment:
Should we treat this the same as a module start section?

Reply:
Agreed, added it in https://github.com/bytecodealliance/wasm-tools/pull/555/commits/44d1d0e04cd7c6ff6fd6779c05b164beeba0a8e6